app/controllers/orders_controller.rb

```diff
@@ -2,29 +2,23 @@
 
 class OrdersController < ApplicationController
   before_action :set_order, only: %i[show edit update destroy]
+  before_action :set_stock, only: %i[new]
+  before_action :authenticate_user!
 
-  # GET /orders or /orders.json
   def index
     @orders = Order.all
   end
 
-  # GET /orders/1 or /orders/1.json
   def show; end
 
-  # GET /orders/new
   def new
-    @order = Order.new
+    @order = Order.new(transaction_type: params[:transaction_type], stock: @stock)
   end
 
-  # GET /orders/1/edit
   def edit; end
 
-  # POST /orders or /orders.json
   def create
-    @order = Order.new(order_params)
-
-    @order.user = current_user
-    @order.stock = Stock.find(order_params[:stock_id])
+    @order = Order.new(order_params.merge(user: current_user))
```

Comment:
Do we want to restrict this here to make sure users can only create orders on their own portfolio? Or we can add this as a separate ticket to make sure we follow up on this if you agree we need it but you don't want to add it now.